server/image_fs_info.go

```diff
@@ -10,24 +10,51 @@ import (
 	types "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
-func getStorageFsInfo(store storage.Store) (*types.FilesystemUsage, error) {
+func getStorageFsInfo(store storage.Store) (*types.ImageFsInfoResponse, error) {
 	rootPath := store.GraphRoot()
+	imagePath := store.ImageStore()
 	storageDriver := store.GraphDriverName()
-	imagesPath := path.Join(rootPath, storageDriver+"-images")
+	graphrootPath := path.Join(rootPath, storageDriver+"-containers")
 
-	bytesUsed, inodesUsed, err := crioStorage.GetDiskUsageStats(imagesPath)
+	if imagePath == "" {
+		graphrootPath = path.Join(rootPath, storageDriver+"-images")
```

Comment:
We can restructure this code a bit in terms of readability. Since you used `GetDiskUsageStats` twice in this function, how about creating a helper function?

```go
func getUsage(path string) (*types.FilesystemUsage, error) {

  bytes, inodes, err := crioStorage.GetDiskUsageStats(path)
  if err != nil {
    return nil, err
  }

  return &types.FilesystemUsage{
    Timestamp: time.Now().UnixNano(),
    FsId: &types.FilesystemIdentifier{Mountpoint: path},
    UsedBytes: &types.UInt64Value{Value: bytes},
    InodesUsed: &types.UInt64Value{Value: inodes},
  }, nil

}
```
```go
func getStorageFsInfo(store storage.Store) (*types.ImageFsInfoResponse, error) {

  rootPath := store.GraphRoot()
  storageDriver := store.GraphDriverName()
  imagePath := store.ImageRoot()
  var graphRootPath string 
  if imagePath == "" {
    graphRootPath = path.Join(rootPath, storageDriver+"-images")  
  } else {
    graphRootPath = path.Join(rootPath, storageDriver+"-containers")
  }

  graphUsage, err := getUsage(graphRoot)
  if err != nil {
    return nil, err
  }

  if imagePath == "" {
    return &types.ImageFsInfoResponse{
      ImageFilesystems: []*types.FilesystemUsage{graphUsage},
      ContainerFilesystems: []*types.FilesystemUsage{graphUsage},  
    }, nil
  }
  resp := &types.ImageFsInfoResponse{
    ContainerFilesystems: []*types.FilesystemUsage{graphUsage},
  }

  if imagePath != "" {

    imageRoot := path.Join(imagePath, storageDriver+"-images")
    
    imageUsage, err := getUsage(imageRoot)
    if err != nil {
      return nil, err
    }

    resp.ImageFilesystems = []*types.FilesystemUsage{imageUsage}

  }

  return resp, nil

}
```

Reply:
Nice suggestion. Will add.

Reply:
Done.